### PR TITLE
Refactoring tests

### PR DESCRIPTION
**Pull request by silverAndroid**
- Implementing async-await to the tests
-  Removing a lot of repetition in object creation and response validation

### CODE REVIEW

**Comment by silverAndroid** [comment]
Just an FYI, [this issue in chai-http](https://github.com/chaijs/chai-http/issues/75) explains why invalid response tests still use callbacks

**Review by jonathanGB — CHANGES_REQUESTED** [review]
changing `user1` to `user` makes reviewing way harder than it needs to be. Should bring it back so we can see only the real changes.

File: test/test.js [hunk 1]
```diff
@@ -156,52 +150,42 @@ describe('Users', () => {
...
+          res.body.data.should.have.property('username').eql(`${user.username}`)
+          res.body.data.should.have.property('datecreated')
+          res.body.data.should.have.property('description')
+          res.body.data.should.have.property('posts')
+          res.body.data.posts.should.be.a('array')
+          res.body.data.posts.length.should.be.eql(0)
+          res.body.data.should.have.property('followers')
+          res.body.data.followers.should.be.a('array')
+          res.body.data.followers.length.should.be.eql(0)
+          res.body.data.should.have.property('following')
```
Comment (jonathanGB) [inline test/test.js]: how about destructuring to get only data?

**Review by jonathanGB — CHANGES_REQUESTED** [review]

File: test/test.js (hunk 1 above)
Comment (silverAndroid) [reply test/test.js]: Can't cause `verifyValidResponse` needs the entire object, unless you mean destructure it after it?

**Comment by silverAndroid** [comment]
Would you prefer if I changed user to user1 for the global variable?

**Comment by jonathanGB** [comment]
yes to user1 global

File: test/test.js (hunk 1 above)
Comment (jonathanGB) [reply test/test.js]: I mean destructuring on line 160

**Comment by BrandonDanis** [comment]
We should look into adding unit tests

File: test/test.js [hunk 2]
```diff
@@ -12,294 +12,214 @@ const should = chai.should()
...
-    AddUser(user, () => {
-      done()
-    })
+  it('POST /users | Should create a new user', async () => {
+    await AddUser(user1)
   })
 
-  it('POST /users | Should not create a user with no username', (done) => {
-    const user = {
+  it('POST /users | Should not create a user with no username', async () => {
```
Comment (jonathanGB) [inline test/test.js]: why async if no use of await?

**Review by jonathanGB — APPROVED** [review]

File: test/test.js (hunk 2 above)
Comment (silverAndroid) [reply test/test.js]: Leftover from when I was trying to implement async await on an invalid response

**Review by BrandonDanis — APPROVED** [review]

**Pull request merged**